### ISSUE TITLE
Reflector/invokeStaticMethod reflection warning

### ISSUE THREAD

**Issue by valerauko** [technomancy/leiningen#2695]
**Initial debugging steps**
Before creating a report, _especially_ around exceptions being thrown when running Leiningen, please check if the error still occurs after:

- [x] Updating to using the latest released version of Leiningen (`lein upgrade`).
- [x] Moving your `~/.lein/profiles.clj` (if present) out of the way. This contains third-party dependencies and plugins that can cause problems inside Leiningen.
- [x] Updating any old versions of plugins in your `project.clj`, especially if the problem is with a plugin not working. Old versions of plugins like nREPL and CIDER (as well as others) can cause problems with newer versions of Leiningen.
- [x] (If you are using Java 9 or newer), updating your dependencies to their most recent versions. Recent JDK's have introduced changes which can break some Clojure libraries.

**Describe the bug**
Reflection warning at
https://github.com/technomancy/leiningen/blob/1dc131128279ef50f21d4d643c4d5c2133c1370d/src/leiningen/run.clj#L78-L80

**To Reproduce**
1. lein new app foo
2. add `:global-vars {*warn-on-reflection* true}` to project.clj
3. lein run will print reflection warning

**Actual behavior**
```
Reflection warning, /tmp/form-init8395418479029823401.clj:1:870 - call to static method invokeStaticMethod on clojure.lang.Reflector can't be resolved (argument types: unknown, java.lang.String, unknown).
```

**Expected behavior**
No reflection warning

**Environment**
- Leiningen Version: Leiningen 2.9.4
- Leiningen installation method: manual
- JDK Version: openjdk version "1.8.0_252"
- OS: Ubuntu 20.04
- Anything else that might be relevant to your problem?

**Additional context**
I noticed that people have attempted to fix this before in #1817 and #2306, but it persisted.

**Comment by psantoro** [comment]
I just hit this issue today when testing an application.  I'm using Leiningen 2.9.8 on Windows 10.  In an attempt to determine if I was doing something wrong, I created a new app called base.  As noted above, :global-vars {*warn-on-reflection* true} must be in your profile for the warning to be displayed.

C:\Users\Peter\work\clojure\base>lein version
Leiningen 2.9.8 on Java 17.0.1 Java HotSpot(TM) 64-Bit Server VM
C:\Users\Peter\work\clojure\base>lein run
**Reflection warning, C:\Users\Peter\AppData\Local\Temp\form-init2790411646534526395.clj:1:926 - call to static method invokeStaticMethod on clojure.lang.Reflector can't be resolved (argument types: unknown, java.lang.String, unknown).**
Hello, World!

As the warning appears to be generated prior to calling main in a brand new app, it doesn't look like I'm doing anything incorrect in my real application.  I wish I could offer more assistance with this, but 'm rather new to clojure.

Peter

**Comment by technomancy** [comment]
Should be fixed in a98bb74f72
